components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Edge.Publisher/src/Engine/DataFlowProcessingEngine.cs

```diff
@@ -159,33 +160,69 @@ public class DataFlowProcessingEngine : IProcessingEngine, IDisposable {
+            diagnosticInfo.EncoderAvgNotificationsMessage = _messageEncoder.AvgNotificationsPerMessage;
+            diagnosticInfo.EncoderAvgIoTMessageBodySize = _messageEncoder.AvgMessageSize;
+            diagnosticInfo.EncoderAvgIoTChunkUsage = chunkSizeAverage;
+            diagnosticInfo.EstimatedIoTChunksPerDay = estimatedMsgChunksPerDay;
+            diagnosticInfo.OutgressBatchBlockBufferSize = _batchNetworkMessageBlock.OutputCount;
+            diagnosticInfo.OutgressInputBufferCount = _sinkBlock.InputCount;
+            diagnosticInfo.OutgressInputBufferDropped = _sinkBlockInputDroppedCount;
+            diagnosticInfo.OutgressIoTMessageCount = _messageSink.SentMessagesCount;
+            diagnosticInfo.ConnectionRetries = _messageTrigger.NumberOfConnectionRetries;
+            diagnosticInfo.OpcEndpointConnected = _messageTrigger.IsConnectionOk;
+            diagnosticInfo.MonitoredOpcNodesSucceededCount = _messageTrigger.NumberOfGoodNodes;
+            diagnosticInfo.MonitoredOpcNodesFailedCount = _messageTrigger.NumberOfBadNodes;
+
+            return diagnosticInfo;
+        }
+
         /// <summary>
         /// Diagnostics timer
         /// </summary>
         /// <param name="state"></param>
         private void DiagnosticsOutputTimer_Elapsed(object state) {
+            var info = GetDiagnosticInfo();
             var totalSeconds = (DateTime.UtcNow - _diagnosticStart).TotalSeconds;
             double totalDuration = _diagnosticStart != DateTime.MinValue ? totalSeconds : 0;
-            double valueChangesPerSec = _messageTrigger.ValueChangesCount / totalDuration;
-            double dataChangesPerSec = _messageTrigger.DataChangesCount / totalDuration;
+            double valueChangesPerSec = info.IngressValueChanges / totalDuration;
+            double dataChangesPerSec = info.IngressDataChanges / totalDuration;
             double valueChangesPerSecLastMin = _messageTrigger.ValueChangesCountLastMinute / Math.Min(totalSeconds, 60d);
             double dataChangesPerSecLastMin = _messageTrigger.DataChangesCountLastMinute / Math.Min(totalSeconds, 60d);
-            double sentMessagesPerSec = totalDuration > 0 ? _messageSink.SentMessagesCount / totalDuration : 0;
-            double messageSizeAveragePercent = Math.Round(_messageEncoder.AvgMessageSize / _maxEncodedMessageSize * 100);
+            double sentMessagesPerSec = totalDuration > 0 ? info.OutgressIoTMessageCount / totalDuration : 0;
+            double messageSizeAveragePercent = Math.Round(info.EncoderAvgIoTMessageBodySize / _maxEncodedMessageSize * 100);
             string messageSizeAveragePercentFormatted = $"({messageSizeAveragePercent}%)";
-            double chunkSizeAverage = _messageEncoder.AvgMessageSize / (4 * 1024);
+            double chunkSizeAverage = info.EncoderAvgIoTMessageBodySize / (4 * 1024);
             double estimatedMsgChunksPerDay = Math.Ceiling(chunkSizeAverage) * sentMessagesPerSec * 60 * 60 * 24;
```

Comment:
`info.EstimatedIoTChunksPerDay` can be used instead of `estimatedMsgChunksPerDay` to not do the same calculation again.

Reply:
changed